### PR TITLE
Send '|deinit' message when trying to hardlink to room aliases

### PR DESCRIPTION
**Pull request by ascriptmaster**
Fixes the problem with room alias "ghost rooms".

### CODE REVIEW

**Comment by Zarel** [comment]
This is a very straightforward solution that I can't think of any disadvantages for.

File: commands.js [hunk 1]
```diff
@@ -701,6 +701,9 @@ var commands = exports.commands = {
 			}
 			return connection.sendTo(target, "|noinit|joinfailed|You do not have permission to join '" + target + "'.");
 		}
+		if (toId(target) in Rooms.aliases) {
+			connection.send(">" + toId(target) + "\n|deinit");
+		}
```
Comment (Zarel) [inline commands.js]: I only have one suggestion, which is to move this above `var joinResult = user.joinRoom(targetRoom, connection);`

Comment (Zarel) [inline commands.js]: The way I'd been trying to get Morf to do it was `|noinit|alias|<actual roomid>` (and have the room replacement be seamless clientside)

Comment (Zarel) [inline commands.js]: But we can save that for later

Comment (ascriptmaster) [inline commands.js]: Doesn't `|noinit` send to the alias as well as the room name? Or am I remembering things wrong

Either way I'll do it when I get home

Comment (Zarel) [inline commands.js]: For now just move `deinit` above `joinRoom`.

**Comment by ascriptmaster** [comment]
`deinit` is above `joinResult` now.

**Pull request merged**